GeoLib/Polygon.cpp

```diff
@@ -38,6 +38,25 @@ Polygon::Polygon(Polygon const& other)
 	}
 }
 
+Polygon& Polygon::operator=(Polygon const& other)
+{
+	if (other._ply_pnts != _ply_pnts) {
+		ERR(
+		    "Polygon::operator=: polygon other and this polygon are not based "
+		    "on the same point vector. Aborting");
+		std::abort();
+	}
+	if (other == *this)
```

Comment:
You compare for equality (i.e., compare the objects), not for identity (i.e., compare their addresses), here. I'd compare `&other == this`. That should also be faster.